packages/react-tabs/Spec.md

```diff
@@ -63,211 +72,270 @@ The existing components are:
 
 #### **Pivot**
 
 A 'Pivot' appears as a horizontal set of borderless buttons with an underline indicating current selection.
-Both v0 and v8 provide a 'Pivot' appearance.
-v8 by default and v0 when underlined and primary props are true.
+
+v0 and v8 provide a 'Pivot' appearance.
+v0: underlined=true
+v8: by default
 
 #### **Block Tab**
 
 A 'Block Tab' appears as a horizontal set of borderless buttons.
 The selection has an active color, rectangular background.
 
 Both v0 and v8 provide the 'Block Tab' appearance.
-v0 by default and v8 when the linkFormat prop equals "tabs".
+v0: by default
+v8: linkFormat="tabs"
 
-#### **Varations**
+#### **Variations**
 
-v0 supports decorating tabs with arrows pointing toward the associated content when the pointing prop is set.
+v0 supports decorating tabs with arrows pointing toward the associated content.
+v0: pointing=true
 
 v8 supports a large tab style with increased padding per tab.
+v8: linkSize="large"
 
 ### Tab Item Content
 
 Both v0 and v8 support text, icon, and text and icon as tab content.
 v0 additionally supports menus as tab content.
 
-Both v0 and v8 support custom rendering of tab content through a render props callbacks per item.
+Both v0 and v8 support custom rendering of tab content through a render props per item.
 
-### Overflow
+### Tab Panel
 
-v0 support overflow when behaving as a toolbar with the accessibility={menuAsToolbarBehavior}.
-This is mutually exclusive with the accessibility={tabListBehavior}.
-
-v8 supports overflow with the overflowBehavor prop of 'menu' or 'none'.
+v0 does not provide a tab panel since it is implemented through the menu component.
+v8 supports the tab panel as the children of PivotItem. The tab content is provided through the header property.
 
 ## Sample Code
 
 ### Default
 
-tabs are stacked horizontally
-tab content is stacked horizontally
-default appearance
+By default tabs are arranged horizontally.
+Content within the tab is arranged horizontally.
+It is centered (horizontally and vertically).
+The default appearance is transparent.
 
 ```tsx
 <TabList>
-  <Tab>One</Tab>
-  <Tab>Two</Tab>
-  <Tab>Three</Tab>
+  <Tab value="tab1">One</Tab>
+  <Tab value="tab2">Two</Tab>
+  <Tab value="tab3">Three</Tab>
 </TabList>
 ```
 
-With Icons
+### Vertical
+
+The vertical prop causes the tabs to be arranged vertically.
+Content within the tab is still arranged horizontally and centered.
 
 ```tsx
-<TabList>
-  <Tab icon={<CheckboxComposite />}>Allowed</Tab>
-  <Tab icon={<BlockedSite />}>Blocked</Tab>
+<TabList vertical>
+  <Tab value="tab1">One</Tab>
+  <Tab value="tab2">Two</Tab>
+  <Tab value="tab3">Three</Tab>
 </TabList>
 ```
 
-### Vertical
+### Vertical Tab content
 
-tabs are stacked vertically, tab content is stacked horizontally
+The verticalTabContent prop causes the content within the tab to be arranged vertically.
+The content is still centered.
+If the icon prop is set, the rendered icon is part of the vertical tab content.
```

Comment:
The problem that I see: once `icon` is not present this variation does not have sense.
It also seems that it's mutually exclusive with `vertical`:

```tsx
<TabList vertical verticalTabContent>
  <Tab></Tab>
</TabList>
```

^ This is nonsense, right?

---

What if we will make it more specific and define on `Tab` itself?

```tsx
<TabList vertical>
  <Tab vertical value="tab1">One</Tab>
  <Tab vertical value="tab2">Two</Tab>
</TabList>
```

Reply:
@GeoffCoxMSFT you mentioned that you replied to this in offline, but I don't see any reply 😺

Why do you think that `verticalTabContent` on `TabList` is better that `vertical` on `Tab`? Are there implementation issues or other motivation?

Reply:
(Here's what I replied, posted again.  My post got stuck in pending state).

The content of the tab may have multiple elements which would be laid out vertically when verticalTabContent is set. If there is an icon then it would be included in the vertical layout. There are also variations of tabs like one with a badge that would have other slot content that could be laid out vertically.

When vertical and verticalTabContent are both set, the layout is similar to the app bar on the left-hand-side of teams or vscode. You can also have vertical where there the tab content is horizontal by default. Both of these layouts are in the figma.

I have the properties on the tab list since the designer asked for consistency across the tabs. I think that mixing and matching vertical and horizontal tab content layout is not likely a common scenario.

It is true that verticalTabContent does not make much visual difference when the tab contains only text. However with the vertical layout the text would be centered instead of aligned left.